### DISCUSSION TITLE
advertisements doesnt alwasy report the same local name

### DISCUSSION THREAD

**Discussion by CosminPerRam** [bluez/bluer#125] [Q&A] [answered]
# Context
I'm essentially using the [gatt_echo_server](https://github.com/bluez/bluer/blob/master/bluer/examples/gatt_echo_server.rs) example running on a Raspberry Pi Zero 2W.

# Problem
When trying to find the device (via [LightBlue](https://punchthrough.com/lightblue/), an app to see BT advertisements or via [react-native-ble-plx](https://github.com/dotintent/react-native-ble-plx), a React Native BT library), the configured local name (`"gatt_echo_server"` in the example) is not always reported and some of the time I find the device's OS host name (which in my case is `raspberry`).

Same goes for a Raspberry Pi 5 that I've also tested on.

# Expected
Always report the local name.

# Debugging
bluetoothd version: 5.55
OS: Debian GNU/Linux 11 (bullseye) aarch64
Kernel: 6.1.21-v8+

I don't really know why this happens only some of the time nor do I know how to properly debug this, any help? Thanks.

**Comment by surban** [comment, marked as answer]
Could you try setting the friendly name using [Adapter::set_alias](https://docs.rs/bluer/latest/bluer/struct.Adapter.html#method.set_alias)?

**Comment by CosminPerRam** [comment]
Thanks for the response.
Unfortunately I will be able to test it and respond only on this monday, will respond then.

**Comment by CosminPerRam** [comment]
Everything seems to be alright, will reopen if problems still occurs, thanks!

**Reply by CosminPerRam** [reply]
A week later update: all seems to be as expected now.
